framework/test/driver/BundleTest.cpp

```diff
@@ -815,6 +816,254 @@ void TestBundleGetPropertyKeys()
   US_TEST_CONDITION_REQUIRED(keys.end() != std::find(keys.begin(), keys.end(), Constants::BUNDLE_ACTIVATOR), "Test that property keys contain the expected keys.");
 }
 
+#if defined (US_BUILD_SHARED_LIBS)
+// Test installing a bundle with invalid meta-data in its manifest.
+void TestBundleManifestFailures()
+{
+  auto f = FrameworkFactory().NewFramework();
+  f.Start();
+  auto bc = f.GetBundleContext();
+
+  std::vector<std::string> validBundleNames({f.GetSymbolicName(), "main"});
```

Comment:
If both `validBundleNames` and the container `names` inside the lambda `validateBundleNames` are `std::set`s instead, then there's no need to sort the containers before calling `std::includes`

Reply:
Good work!